src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -81,6 +82,50 @@ namespace System.Diagnostics
     /// into the output.   This is not likely to be a good idea as it will be very verbose, but is useful to quickly
     /// discover what is available.
     /// 
+    /// 
+    /// * How data is loged in the EventSource 
+    /// 
+    /// By default all data from Diagnostic sources is logged to the the DiagnosticEventSouce event called 'Event' 
+    /// which has three fields  
+    /// 
+    ///     string SourceName, 
+    ///     string EventName, 
+    ///     IEnumerable<KeyValuePair<string, string>> Argument
+    /// 
+    /// However to spport start-stop activity tacking, there are six other events that can be used 
+    /// 
+    ///     Activity1Start         
+    ///     Activity1Stop
+    ///     Activity2Start
+    ///     Activity2Stop
```

Comment:
Two is not necessarily the right number, but it covers a lot of ground.  Basically you can reuse Activity1 for all activities that for do not overlap in time (that is they don't nest).    Those that nest need Activity2 (or more), however you are only likely to need a handful.  One of these was declared to be allowed to be recursive (RecActivity1).

If we need to add more, it is easy enough (simply add them, the code lights up if they are declared), but by far the greatest value is being able to declare a top-level activity.